libs/gl-client/src/lnurl/pay/mod.rs

```diff
@@ -123,6 +173,47 @@ fn ensure_amount_is_within_range(
     Ok(())
 }
 
+//LUD-16: Paying to static internet identifiers.
+pub fn parse_lightning_address(lightning_address: &str) -> Result<String> {
+    let lightning_address_components: Vec<&str> = lightning_address.split("@").collect();
+
+    let username = lightning_address_components.get(0);
+
+    if username.is_none() {
+        return Err(anyhow!("Could not parse username in lightning address"));
+    }
+
+    let path = username.unwrap();
+
+    let domain = lightning_address_components.get(1);
+
+    if domain.is_none() {
+        return Err(anyhow!("Could not parse domain in lightning address"));
+    }
+
+    let domain = domain.unwrap();
```

Comment:
Patternmatching allows you not to use `unwrap()` in turn ensuring the check above and the unwrap don't get accidentally separated.

```
let domain = match domain {
  Some(d) => d.
  None => return Err(anyhow!("no domain"))
};
```

Reply:
Changed to use pattern matching